pkg/opensearch/response_parser.go

```diff
@@ -86,29 +87,187 @@ func (rp *responseParser) getTimeSeries(configuredFields es.ConfiguredFields) (*
 			Frames: data.Frames{},
 		}
 
-		switch target.Metrics[0].Type {
-		case rawDataType:
-			queryRes = processRawDataResponse(res, configuredFields, queryRes)
-		case rawDocumentType:
-			queryRes = processRawDocumentResponse(res, target.RefID, queryRes)
-		case logsType:
-			queryRes = processLogsResponse(res, configuredFields, queryRes)
-		default:
-			props := make(map[string]string)
-			err := rp.processBuckets(res.Aggregations, target, &queryRes, props, 0)
-			if err != nil {
-				return nil, err
+		// trace span condition
+		if target.luceneQueryType == "Traces" {
+			queryRes = processTraceSpansResponse(res, queryRes)
+		} else {
+			switch target.Metrics[0].Type {
+			case rawDataType:
+				queryRes = processRawDataResponse(res, configuredFields, queryRes)
+			case rawDocumentType:
+				queryRes = processRawDocumentResponse(res, target.RefID, queryRes)
+			case logsType:
+				queryRes = processLogsResponse(res, target.Metrics[0].Settings.Get("limit").MustString(), configuredFields, queryRes)
+			default:
+				props := make(map[string]string)
+				err := rp.processBuckets(res.Aggregations, target, &queryRes, props, 0)
+				if err != nil {
+					return nil, err
+				}
+				rp.nameFields(&queryRes.Frames, target)
+				rp.trimDatapoints(&queryRes.Frames, target)
 			}
-			rp.nameFields(&queryRes.Frames, target)
-			rp.trimDatapoints(&queryRes.Frames, target)
-		}
 
+		}
 		result.Responses[target.RefID] = queryRes
+
 	}
 	return result, nil
 }
 
-func processLogsResponse(res *es.SearchResponse, configuredFields es.ConfiguredFields, queryRes backend.DataResponse) backend.DataResponse {
+func processTraceSpansResponse(res *es.SearchResponse, queryRes backend.DataResponse) backend.DataResponse {
+	propNames := make(map[string]bool)
+	docs := make([]map[string]interface{}, len(res.Hits.Hits))
+
+	for hitIdx, hit := range res.Hits.Hits {
+		var withKeysToObj map[string]interface{}
+		if hit["_source"] != nil {
+			// some k:v pairs come from OpenSearch with field names in dot notation: 'span.attributes.http@status_code': 200,
+			// namely TraceSpanRow.Attributes and TraceSpanRow.Resource
+			// this turns everything into maps we can index and access
+			withKeysToObj = utils.FlattenNestedFieldsToObj(hit["_source"].(map[string]interface{}))
+		}
+
+		doc := map[string]interface{}{
+			"_id":     hit["_id"],
+			"_type":   hit["_type"],
+			"_index":  hit["_index"],
+			"_source": withKeysToObj,
+		}
+
+		for k, v := range withKeysToObj {
```

Comment:
is this missing `serviceName`?

As a style nit, I think you could do a `switch` on the value of `k`

Reply:
The `serviceName` field stays the same, as it comes from OpenS in the correct string we need and we don't need to remap it.